docs/RelaxNGSupport.md

```diff
@@ -0,0 +1,131 @@
+
+ * [RelaxNG XML syntax](https://relaxng.org/tutorial-20011203.html): you will benefit the same support than XML, XSD (completion, hover, validation, etc).
+ 
+![RelaxNG XML Syntax Overview](images/RelaxNG/RelaxNGXMLSyntaxOverview.png)
+  
+ * [RelaxNG Compact syntax](https://relaxng.org/compact-tutorial-20030326.html) you will benefit only with syntax coloration.
+ 
+![RelaxNG Compact Syntax Overview](images/RelaxNG/RelaxNGCompactSyntaxOverview.png)
+
+# Validation
+
+XML validation based on RelaxNG (rng, rnc) is supported:
+
+![RelaxNG Validation](images/RelaxNG/RelaxNGValidation.png)
+
+# Completion
+
+XML completion based on RelaxNG (rng, rnc) is supported. The completion for rng can show the documentation:
+
+![RelaxNG Completion And Documentation](images/RelaxNG/RelaxNGCompletionAndDocumentation.png)
+
+# Hover
+
+Hover based on RelaxNG rng can show the documentation:
+
+![RelaxNG Hover And Documentation](images/RelaxNG/RelaxNGHoverAndDocumentation.png)
+
+# Go to Type Definition
+
+From the XML you can go to the type definition to navigate to the element/attribute declaration for rnc, rng both.
+
+To do that, select an XML element/attribute and use the contextual meny `Go to Type Definition`: 
+
+![Go to Type Definition Menu](images/RelaxNG/GoToTypeDefinitionMenu.png)
+
+The click on this menu item, will open the rnc, rng grammar file and set the cursor to the proper element/attribute declaration:
+
+![Go to Type Definition result](images/RelaxNG/GoToTypeDefinitionResult.png)
+
+# How to start?
+
+## Using snippets
+
+You can use snippets to create quickly :
+
+ * an XML file associate with a RelaxNG grammar
+ * a RelaxNG XML file
+ 
+Here a demo:
+
+![RelaxNG snippets support](images/RelaxNG/RelaxNGSnippetsSupport.gif)
+
+## Creating XML, rng, rnc files
+
+Create RelaxNG grammar with Compact syntax `addressBook.rnc` file like this: 
+
+```rnc
+element addressBook {
+  element card {
+    element name { text },
+    element email { text }
+  }*
+} 
+```
+
+Create RelaxNG grammar with XML syntax `addressBook.rng` file like this: 
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<grammar xmlns="http://relaxng.org/ns/structure/1.0">
+    <start>
+        <element name="addressBook">
+            <zeroOrMore>
+                <element name="card">
+                    <ref name="cardContent"/>
+                </element>
+            </zeroOrMore>
+        </element>
+    </start>
+    <define name="cardContent">
+        <element name="name">
+            <text/>
+        </element>
+        <element name="email">
+            <text/>
+        </element>
+    </define>
+</grammar>
+```
+
+Create the XML `addressBook.xml` file to validate with `addressBook.rng` like this: 
+
+```xml
+<?xml-model href="addressBook.rng" type="application/xml"?>
+<addressBook>   
+  <card>
+    <name></name>
+    <email></email>
+  </card>
+</addressBook> 
+```
+
+Now you can enjoy with RelaxNG support like [this demo](#relaxng-support).

```

Comment:
If you change the text before the example XML the way I suggested, you can remove this sentence.

Reply:
removed